### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by AntoninaMJ**
Added returning ResponseEntity for all @PostMapping in controllers
Added integration tests (except tests for reward and taskToDo)
Added @ResponseStatus(HttpStatus.NOT_FOUND) in somethingNotExistsException

### CODE REVIEW

File: src/main/java/pl/antonina/tasks/child/ChildController.java [hunk 1]
```diff
@@ -33,8 +36,15 @@ public ChildView getChild(@ApiIgnore Principal parentPrincipal, @PathVariable lo
     }
 
     @PostMapping
-    public void addChild(@ApiIgnore Principal parentPrincipal, @Validated @RequestBody ChildData childData) {
-        childService.addChild(parentPrincipal, childData);
+    public ResponseEntity<Void> addChild(@ApiIgnore Principal parentPrincipal, @Validated @RequestBody ChildData childData) {
```
Comment (malpiszon) [inline src/main/java/pl/antonina/tasks/child/ChildController.java:39]: Please check if it works properly. There can be problems when using Void here.

File: src/main/java/pl/antonina/tasks/child/ChildController.java [hunk 2]
```diff
@@ -33,8 +36,15 @@ public ChildView getChild(@ApiIgnore Principal parentPrincipal, @PathVariable lo
...
+        long childId = childService.addChild(parentPrincipal, childData);
+
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .path("/{childId}")
+                .buildAndExpand(childId)
+                .toUri();
```
Comment (malpiszon) [inline src/main/java/pl/antonina/tasks/child/ChildController.java]: Moving it to a separated method would make this method a lot more readable! :)

File: src/test/java/pl/antonina/tasks/child/ChildCreator.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.UUID;
+
+class ChildCreator {
+    static ChildData createChildData() {
+        ChildData childData = new ChildData();
+        childData.setGender(Gender.FEMALE);
+        childData.setName("Name");
+        childData.setBirthDate(LocalDate.of(2017, 02, 02));
+        UserData userData = new UserData();
+        final String randomEmail = UUID.randomUUID().toString() + "@gmail.com";
```
Comment (malpiszon) [inline src/test/java/pl/antonina/tasks/child/ChildCreator.java:17]: Probably not needed here, but this is interesting project: https://github.com/DiUS/java-faker

File: src/main/java/pl/antonina/tasks/child/ChildController.java (hunk 1 above)
Comment (AntoninaMJ) [reply src/main/java/pl/antonina/tasks/child/ChildController.java:39]: I checked and it's working. I don't need to return any body.

File: src/test/java/pl/antonina/tasks/child/ChildCreator.java (hunk 3 above)
Comment (AntoninaMJ) [reply src/test/java/pl/antonina/tasks/child/ChildCreator.java:17]: ok, I added a new issue, after this PR I will check it.

File: src/main/java/pl/antonina/tasks/util/LocationUtils.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package pl.antonina.tasks.util;
+
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+
+public class LocationUtils {
+
+    public static URI getLocation(long id) {
```
Comment (malpiszon) [inline src/main/java/pl/antonina/tasks/util/LocationUtils.java:9]: In general it's good approach, but static methods are difficult to mock, so it's always worth to remember about this! :)

**Review by malpiszon — APPROVED** [review]

**Review by malpiszon — APPROVED** [review]

**Pull request merged**